ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +265,64 @@ const createSecret = async (res, appid, payload, DB) => {
     }
 }
 
+// Queries all secrets created by your appid. Maximum 4000 line 
+const listSecrets = async (res, appid, payload, DB) => {
+    try {
+        // get and check parameter 
+        let secretName = getParam_String(payload, 'secretName')
+        if (!checkStringParam(secretName, false, SECRETNAME_LENGTH_MAX)) {
+            res.send(_result(400, `secretName must be string and length not more than ${SECRETNAME_LENGTH_MAX}`))
+            return
+        }
+
+        // build selector for search secret_metadata
+        let selector = {
+            appid: appid
+        }
+        if (secretName != '' && secretName != undefined) {
```

Comment:
whether should query deleted flag? deletedflag only use for tag or can be used on secretname?

Reply:
The function just return meta info, so logic delete secret should be return, and result has plannedDeleteTime

Reply:
This is the question I want to know. we can note and discuss.